protected/humhub/libs/Helpers.php

```diff
@@ -179,27 +180,61 @@ public static function GetUniqeId()
     /**
      * Checks if the class has this class as one of its parents
      *
-     * @param string $className
-     * @param string $type
-     * @return boolean
+     * @param string|object $className Object or classname to be checked
+     * @param string|array $type
+     * @param bool $throw
+     * @param string|null $matchingClass
+     *
+     * @return string|null
      */
-    public static function CheckClassType($className, $type = '')
+    public static function checkClassType($className, $type, bool $throw = true, ?string &$matchingClass = null): ?string
```

Comment:
Can you please add more details of the usage of the `$matchingClass` parameter?

Reply:
Happy to do so. Hoever, I'm not currently using this parameter anywhere. So I may as well remove it and we re-add it when reqired. It was more a spontaneous idea, since the variable already existed in the code of the function.

It's a bit like the `$matches` parameter of `preg_match`. I thought it might be useful to know straight away which class is implemented. So one could do somethign along the lines of (a fabricated example!)

```php
$className = checkClassType($className , [ContentAddonActiveRecord::class, ContentActiveRecord::class], true, $matchingClass);

switch($matchingClass) {
    case ContentAddonActiveRecord::class:
            /** @var ContentAddonActiveRecord $object */
            return $object->canEdit($userId) || $object->content->canEdit($userId);

    case ContentActiveRecord::class:
            /** @var ContentActiveRecord $object */
            return $object->content->canEdit($userId);
}
```

rather than

```php
$className = checkClassType($className , [ContentAddonActiveRecord::class, ContentActiveRecord::class]);

if (is_a($className, ContentAddonActiveRecord::class)) {
	/** @var ContentAddonActiveRecord $object */
	return (new $className())->canEdit($userId);
}

if (is_a($className, ContentActiveRecord::class)) {
	/** @var ContentActiveRecord $object */
	return (new $className())->content->canEdit($userId);
}
```

Reply:
Ok, I see, that's an elegant solution.

If there is some example usage in the core, you are welcome to leave it in.

Reply:
> ... However, I'm not currently using this parameter anywhere. (...) It was more a spontaneous idea, since the variable already existed in the code of the function.

No, as mentioned above, I'm not currently using it. So I'll remove the parameter for now and anyone who might use it can re-add the parameter. 👍